packages/router/src/__tests__/useBlocker.test.tsx

```diff
@@ -0,0 +1,100 @@
+import { renderHook, act } from '@testing-library/react'
+import { describe, it, expect, vi } from 'vitest'
+
+import { gHistory, navigate } from '../history'
+import { useBlocker } from '../useBlocker'
+
+describe('useBlocker', () => {
```

Comment:
These tests are all great, and normally this would be all we asked for.

But for this change in particular I'm still nervous about someone on the team breaking the functionality in the future without noticing. Especially since a unit test can't test what this hook *really* does, which is stops browser navigation. 

Therefore I'm going to ask you to also add a test for this to our Playwright tests. Let me know if you're not interested in doing that, and I'll get someone on the team to help you.

But if you do want to take that on too I'd recommend adding `useBlocker()` to `__fixtures__/test-project/web/src/pages/ContactUsPage/ContactUsPage.tsx`. You can then add a new `contactUs.spec.ts` file in `tasks/smoke-tests/serve/tests` with a test that makes sure navigation is blocked if there are unsaved changes in the contact form.
(The other two spec files in there both run tests actually specified in files inside `shared/`, but for this one I think just running it for `serve` (and not both `dev` and `serve`) is good enough, so you can just put the full test inside that one `contactUs.spec.ts` file)

Thanks! 🙏